Redirecting to page user was on after login

mrpatrickwest opened the pull request:
Before when someone logged in it would take them to their profile page
no matter what. Instead we want to redirect them to the page that they
were on. By changing the LoginRedirector this should take care of the
problem.

Commit:
Redirecting to page user was on after login

Before when someone logged in it would take them to their profile page
no matter what. Instead we want to redirect them to the page that they
were on. By changing the LoginRedirector this should take care of the
problem.

mrpatrickwest commented:
In order to compare the new files in the third tier will have to do a diff between this file and the one under vitro. I commented out the following code starting at line 72 in LoginRedirector

```
        /*
        if (isSelfEditorWithIndividual()) {
            log.debug("Going to Individual home page.");
            return getAssociatedIndividualHomePage();
        }

        if (!canSeeSiteAdminPage()) {
            log.debug("User not recognized. Going to application home.");
            return getApplicationHomePageUrl();
        }

        if (isLoginPage(afterLoginPage)) {
            log.debug("Coming from /login. Going to site admin page.");
            return getSiteAdminPageUrl();
        } else if (null != afterLoginPage) {
        */
```

And added @Ignore for the test methods exitSelfEditor, exitUnrecognizedSelfEditor and exitDbaFromLoginPage in the test class.

zednis commented:
Do we have a way to test this outside of production?

mrpatrickwest commented:
At the moment, no we do not. Production is the only place we have shibboleth working, so is the only place to really test this out. Definitely something we need to get working in a test environment sooner rather then later.

olyerickson commented:
I agree that constructing a test environment that accurately mimics our
PROD Shib infrastructure would be useful, esp. to give us confidence and
reliability for deployment.

On the other hand, developing it potentially represents a huge distraction.
I recommend we do this very carefully and weigh it against other
priorities.

We also need to look at it in the grand scheme of things: for example,
would it be useful for any of our engagements other than DCO?

John

On Sun, Aug 30, 2015 at 3:39 PM, Patrick West notifications@github.com
wrote:

> At the moment, no we do not. Production is the only place we have
> shibboleth working, so is the only place to really test this out.
> Definitely something we need to get working in a test environment sooner
> rather then later.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/tetherless-world/dco-vivo/pull/16#issuecomment-136177600
> .

## 

John S. Erickson, Ph.D.
Director of Operations, The Rensselaer IDEA
Deputy Director, Web Science Research Center (RPI)
http://tw.rpi.edu olyerickson@gmail.com
Twitter & Skype: olyerickson

mrpatrickwest commented:
We absolutely need to get shibboleth working in a test environment. There have been numerous times where upgrading or changing something in shibboleth needed to be done in a test environment first before doing it in production. Numerous times the shibboleth packages have been upgraded in production only to break login. And it would be nice to be able to poke around different configuration options to make sure that we have the most ideal configuration for our environment. We can't do this in production.

The more distracting event is having to do changes in production only to have it break and spend time trying to fix it. We need this test environment.

Commit:
Comments added to the changes made for return page

Added some comments to explain the change being made to redirect the
user to the page they were on instead of their profile page.

zednis commented:
Is this related to https://jira.tw.rpi.edu/browse/DCO-76 which @am-e is currently working?  Is this pull request still something we are looking to merge or should it be dropped?

mrpatrickwest commented:
It was my first attempt at getting this to work. It didn't work, but I did make changes in support of the redirect being successful.

They had it redirecting to a person's profile page if they are logged in and no redirect link is given and the home page otherwise.

mrpatrickwest commented:
@am-e please use this branch for your work on the redirect issue

am-e commented:
Alright I'll do that.

On Tue, Oct 13, 2015 at 2:17 PM Patrick West notifications@github.com
wrote:

> @am-e https://github.com/am-e please use this branch for your work on
> the redirect issue
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/tetherless-world/dco-vivo/pull/16#issuecomment-147800980
> .
